src/main/java/net/glowstone/entity/GlowLivingEntity.java

```diff
@@ -997,13 +997,23 @@ public boolean entityInteract(GlowPlayer player, InteractEntityMessage message)
         super.entityInteract(player, message);
 
         if (message.getHandSlot() == EquipmentSlot.HAND) {
-            if (!InventoryUtil.isEmpty(player.getItemInHand()) && player.getItemInHand().getType() == Material.LEASH) {
+            ItemStack handUten = player.getItemInHand();
+            if (!InventoryUtil.isEmpty(handUten) && handUten.getType() == Material.LEASH) {
                 if (this.isLeashed() || EventFactory.callEvent(new PlayerLeashEntityEvent(this, player, player)).isCancelled()) {
                     return false;
                 }
 
+                if (handUten.getType() == Material.SADDLE && player.getGameMode() != GameMode.CREATIVE) {
```

Comment:
Is this the proper way to do this? I would've thought that this would be done in the `GlowPig` implementation...